### PR TITLE
Make apt upgrade non-interactive for VM provisioning

### PR DESCRIPTION
**Pull request by gauravIoTEdge**
This PR fixes persistent failures in ubuntu:18.04 jobs.

Currently, the VM provisioning process is unable to run **apt upgrade**, which in turn is because the upgrade of the openssh-server package is conflicting with files that are pre-installed on the VM image.

This change makes the upgrades non-interactive and automatically chooses the newer file(s) in case of any conflict (which is fine since it's a clean VM).

**Testing Notes:**

- Check-in gates[ passed (on my fork)](https://github.com/gauravIoTEdge/iot-identity-service/actions/runs/1890462264)
- Ran [E2E tests (on my fork)](https://github.com/gauravIoTEdge/iot-identity-service/runs/5313085676?check_suite_focus=true) and a couple of 18.04 jobs passed, unlike before.

### CODE REVIEW

File: ci/e2e-tests/test-run.sh
Comment (arsing) [inline ci/e2e-tests/test-run.sh:626]:
```suggestion
            sudo DEBIAN_FRONTEND=noninteractive \
                apt-get \
                -o 'Dpkg::Options::=--force-confnew' \
                -o 'Dpkg::Options::=--force-confdef' \
                -y \
                --allow-downgrades \
                --allow-remove-essential \
                --allow-change-held-packages \
                upgrade
```

**Review by arsing — CHANGES_REQUESTED** [review]

**Comment by arsing** [comment]
>Ran E2E tests and a couple of 18.04 jobs passed, unlike before.

Please share the link.

**Comment by gauravIoTEdge** [comment]
> > Ran E2E tests and a couple of 18.04 jobs passed, unlike before.
> 
> Please share the link.

Link is in testing notes.

**Review by arsing — APPROVED** [review]
Thanks, looks to be fixed. Please get an approval from Gordon too before you apply the ready-to-merge label.

**Review by gordonwang0 — APPROVED** [review]

**Pull request merged**
